include/oneapi/tbb/detail/_concurrent_skip_list.h

```diff
@@ -105,11 +97,13 @@ class skip_list_node {
         return res;
     }
 
-    atomic_node_ptr& atomic_next( size_type level ) {
```

Comment:
May be we can add a new member function: `atomic_next_address`, return `&get_atomic_next` and use it directly in `construct`?
In this case we can avoid using boolean flags just to work around the assertion problem.

Reply:
IMHO both solutions are not that straightforward. 
However extending existing method with a parameter seems to be a lesser evil than introducing new method (with functionality overlapping with existing one, which by itself is misleading).